go/exocom/src/connection/manager.go

```diff
@@ -34,14 +36,18 @@ func (m *Manager) AddWebsocket(socket *websocket.Conn) {
 
 // GetClients returns all the connections
 func (m *Manager) GetClients() (result []Client) {
+	m.servicesMutex.RLock()
 	for role := range m.services {
 		result = append(result, Client{Role: role})
 	}
+	m.servicesMutex.RUnlock()
```

Comment:
why no `defer` here?

Reply:
It's the last meaningful line in the function it shouldn't need a `defer` right?

Reply:
I'm using defer only where its needed (ie, if the return line requires the mutex to still be locked). Otherwise I like scoping the lock to just where it is needed